pkgs/development/python-modules/tensorflow/bin.nix

```diff
@@ -141,73 +141,96 @@ buildPythonPackage {
-      # This is an array containing all the directories in the tensorflow2
-      # package that contain .so files.
-      #
-      # TODO: Create this list programmatically, and remove paths that aren't
-      # actually needed.
-      rrPathArr=(
-        "$out/${python.sitePackages}/tensorflow/"
-        "$out/${python.sitePackages}/tensorflow/core/kernels"
-        "$out/${python.sitePackages}/tensorflow/compiler/mlir/stablehlo/"
-        "$out/${python.sitePackages}/tensorflow/compiler/tf2tensorrt/"
-        "$out/${python.sitePackages}/tensorflow/compiler/tf2xla/ops/"
-        "$out/${python.sitePackages}/tensorflow/include/external/ml_dtypes/"
-        "$out/${python.sitePackages}/tensorflow/lite/experimental/microfrontend/python/ops/"
-        "$out/${python.sitePackages}/tensorflow/lite/python/analyzer_wrapper/"
-        "$out/${python.sitePackages}/tensorflow/lite/python/interpreter_wrapper/"
-        "$out/${python.sitePackages}/tensorflow/lite/python/metrics/"
-        "$out/${python.sitePackages}/tensorflow/lite/python/optimize/"
-        "$out/${python.sitePackages}/tensorflow/python/"
-        "$out/${python.sitePackages}/tensorflow/python/autograph/impl/testing"
-        "$out/${python.sitePackages}/tensorflow/python/client"
-        "$out/${python.sitePackages}/tensorflow/python/data/experimental/service"
-        "$out/${python.sitePackages}/tensorflow/python/framework"
-        "$out/${python.sitePackages}/tensorflow/python/grappler"
-        "$out/${python.sitePackages}/tensorflow/python/lib/core"
-        "$out/${python.sitePackages}/tensorflow/python/lib/io"
-        "$out/${python.sitePackages}/tensorflow/python/platform"
-        "$out/${python.sitePackages}/tensorflow/python/profiler/internal"
-        "$out/${python.sitePackages}/tensorflow/python/saved_model"
-        "$out/${python.sitePackages}/tensorflow/python/util"
-        "$out/${python.sitePackages}/tensorflow/tsl/python/lib/core"
-        "$out/${python.sitePackages}/tensorflow.libs/"
-        "${rpath}"
-      )
-
-      # The the bash array into a colon-separated list of RPATHs.
-      rrPath=$(IFS=$':'; echo "''${rrPathArr[*]}")
-      echo "about to run patchelf with the following rpath: $rrPath"
-
-      find $out -type f \( -name '*.so' -or -name '*.so.*' \) | while read lib; do
-        echo "about to patchelf $lib..."
-        chmod a+rx "$lib"
-        patchelf --set-rpath "$rrPath" "$lib"
-        ${lib.optionalString cudaSupport ''
-          addDriverRunpath "$lib"
-        ''}
-      done
-    '';
+    # When using the cpu-only wheel, the final package will be named `tensorflow_cpu`.
+    # Then, in each package requiring `tensorflow`, our pythonRuntimeDepsCheck will fail with:
+    # importlib.metadata.PackageNotFoundError: No package metadata was found for tensorflow
+    # Hence, we manually rename the package to `tensorflow`.
+    lib.optionalString (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86_64 && (!cudaSupport))
```

Comment:
Why make this conditional? Is it unlikely that other platforms will get CPU support going forward?

Reply:
On other platforms, the wheel is already named `tensorflow`. Only the wheel for `x86_64-linux` is called `tensorflow_cpu`.

Reply:
That is because x86-64-linux is at this time the only platform with CPU support.

Why is `!cudaSupport` insufficient?

Reply:
Well, if you would have only `lib.optionalString (!cudaSupport)`, then the patching would occur on `aarch64-linux` (without enabling cuda support).
However, there is no `tensorflow_cpu` folder to move. It is already called `tensorflow` and thus the patch would fail:
```
mv: cannot stat 'tensorflow_cpu-2.18.0.dist-info': No such file or directory
```

The only case where you have a `tensorflow_cpu` to rename to `tensorflow` is when:
- You are on `x86_64-linux`
- cuda support is disabled (`cudaSupport = false`)

Reply:
I'm confused by the thread, but you could also just make the bash part more generic.
E.g. if you could test if `tensorflow-"$version".dist-info` already exists and exit the branch early.

Also I've no idea what the compatibility story, but from the comprehensibility PoV I think scoping like `(cd ...; do stuff)` is obviously clearer than `pushd ... ; do stuff; popd`

Reply:
The idea is something like `if match "/cpu/? src.url`. Like do an exact match for when we use CPU, without tying it to unrelated factors.

Reply:
Very good idea ! Done.

Reply:
I have adjusted the regex to `".*tensorflow_cpu.*"` because yours was not matching.